x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -494,4 +494,47 @@ export class ManifestManager {
   public getArtifactsClient(): EndpointArtifactClientInterface {
     return this.artifactClient;
   }
+
+  /**
+   * Cleanup .fleet-agents indice if there are some orphan artifacts
+   */
+  public async cleanup(manifest: Manifest) {
+    try {
+      const fleetArtifacts = await this.artifactClient.listArtifacts();
+
+      if (isEmpty(fleetArtifacts)) {
+        return;
+      }
+
+      const artifactsToDelete = [];
+      const manifestArtifactsIds = manifest
+        .getAllArtifacts()
+        .map((artifact) => getArtifactId(artifact));
+
+      for (const fleetArtifact of fleetArtifacts) {
+        const artifactId = getArtifactId(fleetArtifact);
+        const isArtifactInManifest = manifestArtifactsIds.includes(artifactId);
+
+        if (!isArtifactInManifest) {
+          this.logger.info(`Artifact ${artifactId} needs to be removed because it's orphan`);
+          artifactsToDelete.push(this.artifactClient.deleteArtifact(artifactId));
+        }
+      }
+
+      if (isEmpty(artifactsToDelete)) {
+        return;
+      }
+
+      this.logger.info(
+        `Starting .fleet-agents cleanup, ${artifactsToDelete.length} orphan artifactes are going to be removed`
```

Comment:
Suggestion:  Capture the artifacts that will be deleted and then output them here. the ID of the artifact will not provide us much from a debugging standpoint, so I'm thinking that we actually log the entire artifact (maybe minus the `body` (content)) so that we have more insight into what type of artifact was found to be orphaned. 

So the message to the `info` above could be: 
```suggestion
        `Cleaning up ${badArtifacts.length} orphan artifacts`
```
(also: I would suggest not mentioning "fleet" as that might lead a client to believe there is something wrong with Fleet - had that happen to me 🤦‍♂️  )


Q. should this be an `error()` call instead? if so, then convert the message to an EndpointError and attach the actual artifacts to it (actually: you might want to do this anyway with the call to `.info()`, as in:

```typescript
this.logger.error(
  new EndpointError(`Cleaning up ${badArtifacts.length} orphan artifacts`, badArtifacts)
)
```

We could also dump them via `.debug()`, but then we would only ever be able to retrieve that data if the client is running kibana with `verbose` set to true.

Reply:
I'm not sure if this should be an error message but I'm fine doing this change. Also, the logs inside the catch need to be an EndpointError too?

Reply:
Yeah, probably not an error because its not actually causing an error. maybe `warn()`?